compiler/rustc_symbol_mangling/src/typeid/typeid_itanium_cxx_abi.rs

```diff
@@ -1112,8 +1125,48 @@ pub fn typeid_for_instance<'tcx>(
     mut instance: Instance<'tcx>,
     options: TypeIdOptions,
 ) -> String {
-    if matches!(instance.def, ty::InstanceDef::Virtual(..)) {
-        instance.args = strip_receiver_auto(tcx, instance.args)
+    if (matches!(instance.def, ty::InstanceDef::Virtual(..))
+        && instance.def_id() == tcx.lang_items().drop_in_place_fn().unwrap())
+        || matches!(instance.def, ty::InstanceDef::DropGlue(_, Some(_ty)))
+    {
+        // Adjust the type ids of DropGlues
```

Comment:
I think you could skip this whole thing by just setting `instance.args` to `[dyn Drop]` if you want to do what you're doing here exactly, or `[()]` would be equivalent and simpler.

Reply:
Done.